libshaderc/include/shaderc.h

```diff
@@ -31,6 +31,12 @@ typedef enum {
   shaderc_glsl_tess_evaluation_shader
 } shaderc_shader_kind;
 
+typedef enum {
+  shaderc_msg_default           = 0x00,
+  shaderc_msg_spirv_rules       = 0x01,
```

Comment:
It's actually not quite that cut & dried, as glslang has a couple of different OpenGL modes. Currently, if you just pass in EShMsgDefault, it permits using compatibility builtins (e.g. ES2 shaders).

If you pass in EShMsgSpvRules, it uses a more restrictive set of builtins which correspond to more modern GL profiles. This is equivalent to running glslang with -G.

If you pass in EShMsgSpvRules | EShMsgVulkanRules, you can only use the subset which is valid for Vulkan. This is equivalent to running glslang with -V.

If the goal here is to expose target profiles, then we'd need to have 3 options in the enum above. Note that this is a very different goal from what I was aiming to achieve; however, I don't have a fundamental objection if that's the direction the Shaderc maintainers want to take instead.